scripts/upload_results_to_oep.py

```diff
@@ -0,0 +1,143 @@
+    metadata_folder = oem2orm.select_oem_dir(oem_folder_name="metadata")
+
+    # The next command will set up the table. The collect_tables_function collects all metadata
+    # files in a folder and retrives the SQLAlchemy ORM objects and returns them. The Tables are
+    # ordered by foreign key. Having a valid metadata strings is necessary for the following steps.
+    tables_orm = oem2orm.collect_tables_from_oem(db, metadata_folder)
+
+    # create table
+    oem2orm.create_tables(db, tables_orm)
+
+    # Writing data into a table
+    for filename in list_filenames:
+        table_name = os.path.splitext(filename)[0]
+
+        logger.info(f"{filename} is processed")
+
+        data_upload_df = pd.read_csv(
+            os.path.join(filepath, filename), encoding="utf8", sep=";"
+        )
+
+        data_upload_df = data_upload_df.where(pd.notnull(data_upload_df), None)
+
+        # The following command will write the content of your dataframe to the table on the OEP
+        # that was created earlier.<br>
+        # Have a look in the OEP after it ran succesfully!
+        logger.info(f"{filename} is written into table")
+
+        try:
+            data_upload_df.to_sql(
+                table_name,
+                connection=db.engine,
+                schema=SCHEMA,
+                if_exists="append",
+                index=False,
+            )
+
+            logger.info("Inserted data to " + SCHEMA + "." + table_name)
+
+        except Exception as e:
+            logger.error(e)
+            logger.error("Writing to " + table_name + " failed!")
+            logger.error(
+                "Note that you cannot load the same data into the table twice."
+                " There will be an id conflict."
+            )
+            logger.error(
+                "Delete and recreate with the commands above, if you want to test your"
+                " upload again."
+            )
+
+        logger.info(f"{filename} writing into table ended")
+
+        # Writing metadata to the table
+        # Now that we have data in our table it's high time, that we attach our metadata to it.
+        # Since we're using the api, some direct http-requests and a little helper function from
+        # the oep-client, we need to import these new dependencies.
+
+        # We use the metadata folder we set up before. (See the Creating tables section)
+        # If you wan´t to set another folder use the code below:
+
+        # oem_path = oem2orm.select_oem_dir(oem_folder_name="metadata")
+        md_file_name = f"{filename}.json"
+
+        # First we're reading the metadata file into a json dictionary.
+        logger.info(f"{filename} read metadata")
+
+        metadata = oem2orm.mdToDict(
+            oem_folder_path=metadata_folder, file_name=md_file_name
+        )
+
+        # Then we need to validate the metadata.
+        # logger.info(f'{file} metadata validation')
+        oem2orm.omi_validateMd(metadata)
```

Comment:
Can the metadata be validated before creating the tables and uploading the data?

Reply:
This does not work ATM, but I am working on it, but there are some hurdles to making this official. 

https://github.com/OpenEnergyPlatform/omi/issues/62#issuecomment-1263582600
https://github.com/OpenEnergyPlatform/omi/pull/63